frontend/src/component/Activity/Preview.jsx

```diff
@@ -0,0 +1,232 @@
+import React from 'react'
+import { translate } from 'react-i18next'
+import { Link } from 'react-router-dom'
+import {
+  buildFilePreviewUrl as jpgPreviewUrl,
+  removeExtensionOfFilename,
+  removeInteractiveContentFromHTML,
+  CONTENT_TYPE,
+  SCREEN_SIZE
+} from 'tracim_frontend_lib'
+import { FETCH_CONFIG, PAGE } from '../../util/helper.js'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import { getHTMLPreview } from '../../action-creator.async.js'
+
+require('./Preview.styl')
+
+// INFO - RJ - 2020-11-17 - possible preview sizes with their media queries, ordered from smallest to biggest
+const PREVIEW_WIDTHS = [
+  [
+    `(max-width: ${SCREEN_SIZE.MAX_SM})`,
+    376 // 400px - margins (10 * 2px) - borders (2 * 1px)
+  ],
+  [
+    '',
+    776 // 800px - margins (10 * 2px) - borders (2 * 1px)
+  ]
+]
+
+const MAX_PREVIEW_HEIGHT = 300
+
+class Preview extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      previewComponent: this.isHtmlPreview() ? null : this.getJPEGPreview(),
+      previewOverflow: false,
+      previewLoading: this.isHtmlPreview(),
+      previewUnavailable: false
+    }
+  }
+
+  isHtmlPreview () {
+    const type = this.props.content.content_type
+    return type === CONTENT_TYPE.HTML_DOCUMENT || type === CONTENT_TYPE.THREAD
+  }
+
+  handleUnavailablePreview = () => {
+    this.setState({
+      previewLoading: false,
+      previewComponent: (
+        <>
+          <i className='fa fa-eye-slash' />
+          <span>{this.props.t('No preview available')}</span>
+        </>
+      ),
+      previewUnavailable: true
+    })
+  }
+
+  setLoadingPreview () {
+    this.setState({ previewLoading: true })
+  }
+
+  async getHTMLPreviewCode (content) {
+    if (content.content_type === CONTENT_TYPE.HTML_DOCUMENT) {
+      return content.raw_content
+    }
+
+    const fetchResultGetHTMLPreview = await getHTMLPreview(
+      content.workspace_id,
+      content.content_type,
+      content.content_id,
+      content.label
+    )
+
+    if (!fetchResultGetHTMLPreview.ok) {
+      return null
+    }
+
+    return fetchResultGetHTMLPreview.text()
+  }
+
+  async getHTMLPreviewComponent () {
+    const { content } = this.props
+
+    const htmlCode = await this.getHTMLPreviewCode(content)
+
+    if (htmlCode === null) {
+      // RJ - NOTE - 2020-11-18: comparing to null, since the empty string
+      // would be a valid HTML preview code
+
+      this.handleUnavailablePreview()
+      return null
+    }
+
+    return (
+      <div className='activityFeed__preview__html'>
+        <article
+          dangerouslySetInnerHTML={{
+            __html: removeInteractiveContentFromHTML(htmlCode)
+          }}
+        />
+      </div>
+    )
+  }
+
+  getJPEGPreview (previewUrl) {
+    const { content } = this.props
+    const filenameNoExtension = removeExtensionOfFilename(content.filename)
+    const FIRST_PAGE = 1
+
+    const prev = (width) => (
```

Comment:
I would prefer a more explicit function name

Reply:
renamed to previewURL